--- a/svc_inference.py
+++ b/svc_inference.py
@@ -117,7 +117,7 @@
             sub_har = source[:, :, cut_s *
                              hop_size:cut_e * hop_size].to(device)
             sub_out = model.inference(sub_ppg, sub_pit, spk, sub_len, sub_har)
-            sub_out = sub_out[0, 0].data.cpu().detach().numpy()
+            sub_out = sub_out[0, 0].data.float().cpu().detach().numpy()
 
             sub_out = sub_out[cut_s_out:cut_e_out]
             out_audio.extend(sub_out)
@@ -135,7 +135,7 @@
             sub_len = torch.LongTensor([all_frame - cut_s]).to(device)
             sub_har = source[:, :, cut_s * hop_size:].to(device)
             sub_out = model.inference(sub_ppg, sub_pit, spk, sub_len, sub_har)
-            sub_out = sub_out[0, 0].data.cpu().detach().numpy()
+            sub_out = sub_out[0, 0].data.float().cpu().detach().numpy()
 
             sub_out = sub_out[cut_s_out:]
             out_audio.extend(sub_out)
