--- a/svc_inference.py
+++ b/svc_inference.py
@@ -124,7 +124,7 @@
                              hop_size:cut_e * hop_size].to(device)
             sub_out = model.inference(
                 sub_ppg, sub_vec, sub_pit, spk, sub_len, sub_har)
-            sub_out = sub_out[0, 0].data.cpu().detach().numpy()
+            sub_out = sub_out[0, 0].data.float().cpu().detach().numpy()
 
             sub_out = sub_out[cut_s_out:cut_e_out]
             out_audio.extend(sub_out)
--- a/crepe/decode.py
+++ b/crepe/decode.py
@@ -66,7 +66,7 @@
         probs = torch.nn.functional.softmax(logits, dim=1)
 
     # Convert to numpy
-    sequences = probs.cpu().numpy()
+    sequences = probs.float().cpu().numpy()
 
     # Perform viterbi decoding
     bins = np.array([
